### PR TITLE
test: update power/notification specs to expect

### PR DESCRIPTION
**Pull request by codebytere**
##### Description of Change
This PR updates the `notification`, `power-monitor`, and `power-save-monitor` specs as part of ongoing work to migrate our specs from `assert` to chai's `expect` with `dirty-chai` for better error messages.

/cc @alexeykuzmin

##### Checklist
- [x] PR description included and stakeholders cc'd
- [x] `npm test` passes
- [x] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [x] commit messages or PR title follow semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)

##### Release Notes
Notes: no-notes

### CODE REVIEW

File: spec/api-power-monitor-spec.js [hunk 1]
```diff
@@ -27,9 +31,7 @@ xdescribe('powerMonitor', () => {
       reset = Promise.promisify(logindMock.Reset, {context: logindMock})
     })
 
-    after(async () => {
-      await reset()
-    })
+    after(async () => { await reset() })
```
Comment (alexeykuzmin) [inline spec/api-power-monitor-spec.js]: Just pass `reset` to the `after()` call.
`reset` returns a `Promise`, and `after` can will properly handle it.

File: spec/api-notification-spec.js [hunk 2]
```diff
@@ -81,10 +85,10 @@ describe('Notification module', () => {
       }
     ]
 
-    assert.equal(n.actions[0].type, 'button')
-    assert.equal(n.actions[0].text, '3')
-    assert.equal(n.actions[1].type, 'button')
-    assert.equal(n.actions[1].text, '4')
+    expect(n.actions[0].type).to.equal('button')
```
Comment (alexeykuzmin) [inline spec/api-notification-spec.js]: Shouldn't we check that `n.actions` is an array of the expected size first?

File: spec/api-notification-spec.js [hunk 3]
```diff
@@ -66,10 +70,10 @@ describe('Notification module', () => {
       ]
     })
 
-    assert.equal(n.actions[0].type, 'button')
-    assert.equal(n.actions[0].text, '1')
-    assert.equal(n.actions[1].type, 'button')
-    assert.equal(n.actions[1].text, '2')
+    expect(n.actions[0].type).to.equal('button')
```
Comment (alexeykuzmin) [inline spec/api-notification-spec.js]: Shouldn't we check that `n.actions` is an array of the expected size first?

File: spec/api-power-monitor-spec.js [hunk 4]
```diff
@@ -41,16 +43,16 @@ xdescribe('powerMonitor', () => {
...
       logindMock.on('MethodCalled', onceMethodCalled(done))
       // lazy load powerMonitor after we listen to MethodCalled mock signal
       dbusMockPowerMonitor = require('electron').remote.powerMonitor
     })
 
     it('should call Inhibit to delay suspend', async () => {
       const calls = await getCalls()
-      assert.equal(calls.length, 1)
-      assert.deepEqual(calls[0].slice(1), [
+      expect(calls.length).to.equal(1)
```
Comment (alexeykuzmin) [inline spec/api-power-monitor-spec.js]: Array of length 1?

File: spec/api-power-monitor-spec.js [hunk 5]
```diff
@@ -68,16 +70,16 @@ xdescribe('powerMonitor', () => {
...
           dbusMockPowerMonitor.once('resume', () => done())
           emitSignal('org.freedesktop.login1.Manager', 'PrepareForSleep',
             'b', [['b', false]])
         })
 
         it('should have called Inhibit again', async () => {
           const calls = await getCalls()
-          assert.equal(calls.length, 2)
-          assert.deepEqual(calls[1].slice(1), [
+          expect(calls.length).to.equal(2)
```
Comment (alexeykuzmin) [inline spec/api-power-monitor-spec.js]: Array of length 2?

File: spec/api-power-monitor-spec.js [hunk 6]
```diff
@@ -126,35 +128,32 @@ xdescribe('powerMonitor', () => {
     })
 
     describe('powerMonitor.querySystemIdleState', () => {
-      it('notify current system idle state', (done) => {
-        powerMonitor.querySystemIdleState(1, (idleState) => {
-          assert.ok(idleState)
+      it('notify current system idle state', done => {
+        powerMonitor.querySystemIdleState(1, idleState => {
+          expect(idleState).to.be.true()
           done()
```
Comment (alexeykuzmin) [inline spec/api-power-monitor-spec.js]: I've heard that async/await is cool =)

```js
it('notify current system idle state', async () => {
  const idleState = await new Promise(
      resolve => powerMonitor.querySystemIdleState(1, resolve))
  expect(idleState).to.be.true()
})
```

File: spec/api-power-save-blocker-spec.js [hunk 7]
```diff
@@ -1,13 +1,17 @@
...
+chai.use(dirtyChai)
 
 describe('powerSaveBlocker module', () => {
   it('can be started and stopped', () => {
-    assert.equal(powerSaveBlocker.isStarted(-1), false)
+    expect(powerSaveBlocker.isStarted(-1)).to.be.false()
     const id = powerSaveBlocker.start('prevent-app-suspension')
-    assert.ok(id != null)
-    assert.equal(powerSaveBlocker.isStarted(id), true)
+    expect(id).to.not.be.null()
```
Comment (alexeykuzmin) [inline spec/api-power-save-blocker-spec.js]: If it's not `null`, what it is? A number, an object, a symbol?

**Comment by alexeykuzmin** [comment]
@codebytere Sorry for a slow reaction to your PRs =/
You're doing a great job making the tests better )

**Comment by codebytere** [comment]
@alexeykuzmin no worries 😁 i'm just coming back online after PTO so i'm gonna follow up and address these soon!

File: spec/api-power-save-blocker-spec.js (hunk 7 above)
Comment (miniak) [reply spec/api-power-save-blocker-spec.js]: fixed

File: spec/api-power-monitor-spec.js (hunk 6 above)
Comment (miniak) [reply spec/api-power-monitor-spec.js]: It feels forced in this simple case, I would leave it as is

File: spec/api-power-monitor-spec.js (hunk 1 above)
Comment (miniak) [reply spec/api-power-monitor-spec.js]: done

File: spec/api-notification-spec.js (hunk 2 above)
Comment (miniak) [reply spec/api-notification-spec.js]: done

File: spec/api-notification-spec.js (hunk 3 above)
Comment (miniak) [reply spec/api-notification-spec.js]: done

**Comment by miniak** [comment]
@codebytere, @alexeykuzmin I've rebase the branch to fix the conflict with `api-process-spec.js`, which I've already converted before in a separate PR. I've also addressed a few of the comments.

File: spec/api-power-monitor-spec.js (hunk 5 above)
Comment (miniak) [reply spec/api-power-monitor-spec.js]: fixed

File: spec/api-power-monitor-spec.js (hunk 4 above)
Comment (miniak) [reply spec/api-power-monitor-spec.js]: fixed

File: spec/api-power-monitor-spec.js (hunk 6 above)
Comment (alexeykuzmin) [reply spec/api-power-monitor-spec.js]: Ok =/

**Comment by alexeykuzmin** [comment]
Only gn builds failed and only because the branch is relatively old.
Should be good to merge.

**Pull request merged**